### PR TITLE
create annotation classes with sub annotations

### PR DESCRIPTION
**Pull request by simedw**
- support for creating annotations with sub types
- better error message

### CODE REVIEW

File: darwin/dataset/remote_dataset.py [hunk 1]
```diff
@@ -366,35 +367,30 @@ def restore_archived(self, items) -> None:
...
-        name : str
-            The name of the annotation class.
-        type : str
-            The type of the annotation class.
+    def create_annotation_class(self, name: str, type: str, subtypes: List[str] = []):
+        type_ids = []
+        for annotation_type in [type] + subtypes:
+            type_id = self.fetch_annotation_type_id_for_name(annotation_type)
+            if not type_id:
+                list_of_annotation_types = ", ".join([type["name"] for type in self.annotation_types])
```
Comment (FrancescoSaverioZuppichini) [inline darwin/dataset/remote_dataset.py:381]: I have a doubt about this part. So, we are searching for `annotation_type` that may be `type` or something inside `subtypes`. 

Then, if we fail (`fetch_annotation_type_id_for_name` returns `None`) we are going to raise a ValueError saying that specific `annotation_type` was `Unknown` and we add `valid values: {list_of_annotation_types}"`. But if the only `valid values` are the ones inside `list_of_annotation_types` (computed from `self.annotation_types` could we just use the intersection between `subtypes` and `self.annotation_types`?

**Review by FrancescoSaverioZuppichini — COMMENTED** [review]
Great work! Just a little comment

File: darwin/dataset/remote_dataset.py (hunk 1 above)
Comment (simedw) [reply darwin/dataset/remote_dataset.py:381]: But we also want to get all the type ids?

Comment (FrancescoSaverioZuppichini) [reply darwin/dataset/remote_dataset.py:381]: After yesterday discussion, LGTM

**Review by FrancescoSaverioZuppichini — APPROVED** [review]
👍

**Pull request merged**